src/main/java/org/apache/sling/testing/mock/jcr/MockAuthorizable.java

```diff
@@ -67,7 +66,7 @@ abstract class MockAuthorizable implements Authorizable {
                 intermediatePath = "/home/users"; // NOSONAR
             }
         }
-        this.path = Paths.get(intermediatePath, this.id).toString();
+        this.path = intermediatePath + "/" + this.id;
```

Comment:
Sorry about that.  

How about using something like org.apache.jackrabbit.oak.commons.PathUtils.concat(intermediatePath, this.id)  to improve the readability and provide some protection against illegal arguments?

Reply:
yes, this would be better. the class you mention is not available in the classpath.
i assume it's part of `org.apache.jackrabbit:oak-commons`. we could add it, but then we would need to manage the version of this dependency in all contexts as well fitting to the actual oak version used.
so it would be better to find a better methods in the dependencies we already have in place to not over-complicate things here.

Reply:
Wouldn't it be safe to add the extra oak-commons dependency as long as it is using the same (minimum) version as the other oak-* dependencies that already exist in the pom?

Reply:
FYI: I've just proposed https://github.com/apache/sling-org-apache-sling-testing-jcr-mock/pull/21 whose implementation would remove this specific code path.  The users/group would be backed with a real node in the repository rather than the fake path that was used before.  If that PR is approved, then this one should be no longer needed.

Reply:
i've created https://issues.apache.org/jira/browse/SLING-11870 to streamline the oak dependency in jcr-mock, we can approach this after #21 was merged.